sql/src/main/java/io/crate/planner/node/dql/MergePhase.java

```diff
@@ -137,6 +138,49 @@ public static MergePhase sortedMerge(UUID jobId,
         return mergeNode;
     }
 
+    @Nullable
+    public static MergePhase mergePhase(ConsumerContext context,
+                                        Collection<String> executionNodes,
+                                        UpstreamPhase upstreamPhase,
+                                        @Nullable OrderBy orderBy,
+                                        @Nullable List<? extends Symbol> orderBySymbols,
+                                        List<Projection> projections,
+                                        List<Symbol> inputs,
+                                        @Nullable List<DataType> inputTypes,
+                                        boolean isDistributed) {
+        assert !upstreamPhase.executionNodes().isEmpty() : "upstreamPhase must be executed somewhere";
+        if (!isDistributed && upstreamPhase.executionNodes().equals(executionNodes)) {
+            // if the nested loop is on the same node we don't need a mergePhase to receive requests
```

Comment:
@mfussenegger I'm thinking of extracting a new method validateExecutionNodes():

public static void validateExecutionNodes(Collection<String> executionNodes,
                                              Collection<String> upstreamPhaseExecutionNodes,
                                              boolean isDistributed) {
    assert !upstreamPhaseExecutionNodes.isEmpty() : "upstreamPhase must be executed somewhere";
    if (!isDistributed && upstreamPhaseExecutionNodes.equals(executionNodes)) {
        // if the nested loop is on the same node we don't need a mergePhase to receive requests
        // but can access the RowReceiver of the nestedLoop directly
    }
}
Because this shouldn't be called from QueryAndFetchConsumer.
What do you think?